File: starters/couchbase-java-starter/app/controllers/Application.java
package controllers;

import org.ancelin.play2.java.couchbase.Couchbase;
import org.ancelin.play2.java.couchbase.CouchbaseBucket;
import play.*;
import play.libs.F;
import play.libs.Json;
import play.mvc.*;

import views.html.*;

public class Application extends Controller {

    public static Result index() {
        return ok(index.render("Your new application is ready."));
    }

    public static CouchbaseBucket bucket = Couchbase.bucket("default");

    public static class User {
        public String id;
        public String name;
    }

    public static F.Promise<Result> getUser(final String key) {
        return bucket.get(key, User.class).map(new F.Function<User, Result>() {
            @Override
            public Result apply(User user) throws Throwable {
                if (user == null) {
                    return badRequest("Unable to find user with key: " + key);
                }
                return ok(Json.toJson(user));
            }
        });
    }
}

File: attic/starters/couchbase-scala-starter/app/controllers/Application.scala
package controllers

import play.api.mvc._
import play.api.Play.current
import org.ancelin.play2.couchbase.{Couchbase, CouchbaseController}
import play.api.libs.json.JsObject
import models.User
import models.User.userFmt
import org.ancelin.play2.couchbase.plugins.CouchbaseN1QLPlugin._

object Application extends Controller with CouchbaseController {

  implicit val couchbaseExecutionContext = Couchbase.couchbaseExecutor

  def index = Action {
    Ok(views.html.index("Your new application is ready."))
  }

  // Usage of CouchbaseAction
  def getUser(key: String) = CouchbaseAction("default") { bucket =>
    bucket.get[JsObject](key).map { maybeUser =>
      maybeUser
        .map(user => Ok(user))
        .getOrElse(BadRequest(s"Unable to find user with key: $key"))
    }
  }

  // Usage of N1QL plugin
  def findUserByEmail(email: String) = Action.async {
    N1QL( s""" SELECT id, name, email FROM default WHERE email = '${email}' """ ).toList[User].map { users =>
      Ok(users.mkString(" | "))
    }
  }
}

